k6deps does not follow links in archives

Given an archive with the structure below, k6deps does not extract the `main.ts` file which is a link to the `data` file.
Processing the resulting temporary file with `k6pack` fails because this file is missing.

```
[ERROR] Could not resolve "../main.ts"

    ../../../../../../../tmp/k6deps-120020017/file/Users/nobody/code/atypical/pegasus/backend/test/k6/scenarios/commons.ts:2:25:
      2 │ import { BASE_URL } from "../main.ts";
        ╵                          ~~~~~~~~~~~~
```

## structure of archive file

```
-rw-r--r-- 0/0            3726 2025-02-25 18:16 metadata.json
-rw-r--r-- 0/0            4571 2025-02-25 18:16 data
drwxr-xr-x 0/0               0 2025-02-25 18:16 file
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus/backend
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus/backend/test
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus/backend/test-data
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus/backend/test-data/load-tests
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus/backend/test/k6
drwxr-xr-x 0/0               0 2025-02-25 18:16 file/Users/nobody/code/atypical/pegasus/backend/test/k6/scenarios
-rw-r--r-- 0/0           39602 2025-02-21 17:04 file/Users/nobody/code/atypical/pegasus/backend/test-data/load-tests/user_tokens.json
-rw-r--r-- 0/0           14828 2025-02-25 18:15 file/Users/nobody/code/atypical/pegasus/backend/test/k6/config-builder.ts
-rw-r--r-- 0/0            2176 2025-02-25 00:03 file/Users/nobody/code/atypical/pegasus/backend/test/k6/config.ts
h--------- 0/0               0 1970-01-01 01:00 file/Users/nobody/code/atypical/pegasus/backend/test/k6/main.ts link to data
-rw-r--r-- 0/0            4825 2025-02-19 00:45 file/Users/nobody/code/atypical/pegasus/backend/test/k6/scenarios/commons.ts
-rw-r--r-- 0/0            6315 2025-02-24 04:26 file/Users/nobody/code/atypical/pegasus/backend/test/k6/scenarios/practice.ts
-rw-r--r-- 0/0            5832 2025-02-24 04:26 file/Users/nobody/code/atypical/pegasus/backend/test/k6/scenarios/review.ts
-rw-r--r-- 0/0            2186 2025-02-24 04:26 file/Users/nobody/code/atypical/pegasus/backend/test/k6/users.ts
-rw-r--r-- 0/0            1542 2025-02-24 04:26 file/Users/nobody/code/atypical/pegasus/backend/test/k6/utils.ts
```

--- a/archive.go
+++ b/archive.go
@@ -60,11 +60,11 @@ type archiveMetadata struct {
 	Env      map[string]string `json:"env"`
 }
 
-//nolint:forbidigo,gocognit
+const maxFileSize = 1024 * 1024 * 10 // 10M
+
+//nolint:forbidigo
 func extractArchive(dir string, input io.Reader) error {
 	reader := tar.NewReader(input)
-
-	const maxFileSize = 1024 * 1024 * 10 // 10M
 
 	for {
 		header, err := reader.Next()
@@ -87,7 +87,7 @@ func extractArchive(dir string, input io.Reader) error {
 			}
 
 		case tar.TypeReg:
-			if ext := filepath.Ext(target); ext == ".csv" || (ext == ".json" && filepath.Base(target) != "metadata.json") {
+			if shouldSkip(target) {
 				continue
 			}
 
@@ -107,37 +107,52 @@ func extractArchive(dir string, input io.Reader) error {
 		// if it is a link or symlink, we copy the content of the linked file to the target
 		// we assume the linked file was already processed and exists in the directory.
 		case tar.TypeLink, tar.TypeSymlink:
-			if ext := filepath.Ext(target); ext == ".csv" || (ext == ".json" && filepath.Base(target) != "metadata.json") {
+			if shouldSkip(target) {
 				continue
 			}
 
 			linkedFile := filepath.Join(dir, filepath.Clean(filepath.FromSlash(header.Linkname)))
-			source, err := os.Open(filepath.Clean(linkedFile))
-			if err != nil {
-				return err
-			}
-			defer source.Close() //nolint:errcheck
-
-			// we need to get the lined file info to create the target file with the same permissions
-			info, err := source.Stat()
-			if err != nil {
-				return err
-			}
-
-			file, err := os.OpenFile(target, os.O_CREATE|os.O_WRONLY, info.Mode())
-			if err != nil {
-				return err
-			}
-
-			_, err = io.Copy(file, source)
-			if err != nil {
-				return err
-			}
-
-			err = file.Close()
-			if err != nil {
+			if err := followLink(linkedFile, target); err != nil {
 				return err
 			}
 		}
 	}
+}
+
+// indicates if the file should be skipped during extraction
+// we skip csv files and .json except metadata.json
+func shouldSkip(target string) bool {
+	ext := filepath.Ext(target)
+	return ext == ".csv" || (ext == ".json" && filepath.Base(target) != "metadata.json")
+}
+
+//nolint:forbidigo
+func followLink(linkedFile string, target string) error {
+	source, err := os.Open(filepath.Clean(linkedFile))
+	if err != nil {
+		return err
+	}
+	defer source.Close() //nolint:errcheck
+
+	// we need to get the lined file info to create the target file with the same permissions
+	info, err := source.Stat()
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(target, os.O_CREATE|os.O_WRONLY, info.Mode()) //nolint:gosec
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(file, source)
+	if err != nil {
+		return err
+	}
+
+	err = file.Close()
+	if err != nil {
+		return err
+	}
+	return nil
 }

